libraries/AP_AS5600_AOA/AS5600_AOA.cpp

```diff
@@ -0,0 +1,266 @@
+
+extern const AP_HAL::HAL &hal;
+
+/****************************************************
+ * Method: AS5600_AOA
+ * In: none
+ * Out: none
+ * Description: constructor for AMS 5600
+***************************************************/
+ AS5600_AOA::AS5600_AOA()
+{
+
+    bus = 1; //Sets the bus number for the device, unclear what this number should be, trial and error to make it work
+    address = 0x36; //This is the I2C address for the device, it is set by the manufacturer
+
+   /*load register values*/
+
+  _zmco = 0x00;
+  _zpos_hi = 0x01;
+  _zpos_lo = 0x02;
+  _mpos_hi = 0x03;
+  _mpos_lo = 0x04;
+  _mang_hi = 0x05;
+  _mang_lo = 0x06;
+  _conf_hi = 0x07;
+  _conf_lo = 0x08;
+  _raw_ang_hi = 0x0c;
+  _raw_ang_lo = 0x0d;
+  _ang_hi = 0x0e;
+  _ang_lo = 0x0f;
+  _stat = 0x0b;
+  _agc = 0x1a;
+  _mag_hi = 0x1b;
+  _mag_lo = 0x1c;
+  _burn = 0xff;
+
+}
+
+
+void AS5600_AOA::init(){
+
+    dev = hal.i2c_mgr->get_device(bus, address);
+
+    WITH_SEMAPHORE(dev->get_semaphore());
+    dev->set_speed(AP_HAL::Device::SPEED_LOW);
+    dev->set_retries(2);
+
+}
+
+/* mode = 0, output PWM, mode = 1 output analog (full range from 0% to 100% between GND and VDD*/
+void AS5600_AOA::setOutPut(unsigned char mode){
+    unsigned char config_status;
+    config_status = readOneByte(_conf_lo);
+    if(mode == 1){
+        config_status = config_status & 0xcf;
+    }else{
+        config_status = config_status & 0xef;
+    }
+
+    /* Note significant variance from the Arduino AMS_5600 library in this line.
+     *
+     * -> writeOneByte(_conf_lo, lowByte(config_status));
+     *  vs writeOneByte(_conf_lo, config_status); I have removed the lowByte() function
+     *  since it is part of the arduino.h library and I don't have access to it, however,
+     *  since it accesses the lowest byte in the variable, and an unsigned char such as
+     *  config_status only HAS one byte regardless, I think the end result should be identical?
+     */
+    writeOneByte(_conf_lo, config_status);
+}
+
+
+/*******************************************************
+ * Method: getRawAngle
+ * In: none
+ * Out: value of raw angle register
+ * Description: gets raw value of magnet position.
+ * start, end, and max angle settings do not apply
+*******************************************************/
+unsigned short AS5600_AOA::getRawAngle(void)
+{
+  unsigned short angleRaw = readTwoBytes(_raw_ang_hi, _raw_ang_lo);
+
+  unsigned short angleDegrees = angleRaw*0.087;
+
+  AP::logger().Write("AoAR", "Status, TimeUS, Angle", "iQH", int(bool(dev)), AP_HAL::micros64(), angleDegrees);
+
+   return angleDegrees;
+}
+/*******************************************************
+ * Method: highByte
+ * In: Unsigned short
+ * Out: Highest or leftmost byte of the unsigned short
+ * Description: Takes in the unsigned short and returns
+ * the leftmost or highest bite
+*******************************************************/
+
+unsigned char AS5600_AOA::highByte(unsigned short short_in){

```

Comment:
There's a `HIGHBYTE` define available to accomplish the same thing.

Similarly `LOWBYTE`